frontend/src/metabase-lib/lib/queries/NativeQuery.ts

```diff
@@ -533,3 +515,21 @@ export default class NativeQuery extends AtomicQuery {
       });
   }
 }
+
+// look for variable usage in the query (like '{{varname}}').  we only allow alphanumeric characters for the variable name
+// a variable name can optionally end with :start or :end which is not considered part of the actual variable name
+// expected pattern is like mustache templates, so we are looking for something like {{category}} or {{date:start}}
+// anything that doesn't match our rule is ignored, so {{&foo!}} would simply be ignored
+// variables referencing other questions, by their card ID, are also supported: {{#123}} references question with ID 123
+export function recognizeTemplateTags(queryText: string): string[] {
+  const tagNames = [];
+  let match;
+  const re = /\{\{\s*((snippet:\s*[^}]+)|[A-Za-z0-9_\.]+?|#[0-9]*)\s*\}\}/g;
```

Comment:
I'm not familiar with js convention but is it worth it to define this regex outside of this function? 
so we could export it for future usages.

Reply:
Using the regex directly is rather dangerous (note how unreadable it is), hence the idea of reusing this very function `recognizeTemplateTags` instead (it's `export`-ed for that reason).
IOW the regex is just an implementation detail (who knows, we might need to switch to tokenizing SQL in the future).